src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/state_manager_actions.ts

```diff
@@ -145,57 +159,104 @@ export const moveAction = (
     requestedPanelData.column = targetColumn;
     requestedPanelData.row = targetRow;
   }
+  const hasChangedGridSection = targetSectionId !== lastSectionId;
 
   // resolve the new grid layout
   if (
-    hasChangedGridRow ||
+    hasChangedGridSection ||
     !isGridDataEqual(requestedPanelData, lastRequestedPanelPosition.current)
   ) {
     lastRequestedPanelPosition.current = { ...requestedPanelData };
 
-    const nextLayout = cloneDeep(currentLayout);
-    Object.entries(nextLayout).forEach(([rowId, row]) => {
-      const { [interactionEvent.id]: interactingPanel, ...otherPanels } = row.panels;
-      nextLayout[rowId] = { ...row, panels: { ...otherPanels } };
-    });
+    let nextLayout = cloneDeep(currentLayout) ?? {};
+
+    if (!targetSectionId || !nextLayout[targetSectionId]) {
+      // section doesn't exist, so add it
+      const { order: nextOrder } =
+        targetSectionId === 'main-0' ? { order: -1 } : nextLayout[previousSection!];
+
+      // push other sections down
+      Object.keys(nextLayout).forEach((sectionId) => {
+        if (nextLayout[sectionId].order > nextOrder) {
+          nextLayout[sectionId].order += 1;
+        }
+      });
+      // add the new section, which may be renamed by `resolveSections` to `main-<order>`
+      targetSectionId = targetSectionId ?? `main-new`;
+      nextLayout[targetSectionId] = {
+        id: targetSectionId,
+        isMainSection: true,
+        panels: {},
+        order: nextOrder + 1,
+      };
+      requestedPanelData.row = 0;
+    }
+
+    // remove the panel from where it started so that we can apply the drag request
+    delete nextLayout[lastSectionId].panels[activePanel.id];
 
     // resolve destination grid
-    const destinationGrid = nextLayout[targetRowId];
-    const resolvedDestinationGrid = resolveGridRow(destinationGrid, requestedPanelData);
-    nextLayout[targetRowId] = resolvedDestinationGrid;
+    const destinationGrid = nextLayout[targetSectionId] as unknown as GridSectionData;
+    const resolvedDestinationGrid = resolveGridSection(destinationGrid.panels, requestedPanelData);
+    (nextLayout[targetSectionId] as unknown as GridSectionData).panels = resolvedDestinationGrid;
 
     // resolve origin grid
-    if (hasChangedGridRow) {
-      const originGrid = nextLayout[lastRowId];
-      const resolvedOriginGrid = resolveGridRow(originGrid);
-      nextLayout[lastRowId] = resolvedOriginGrid;
+    if (hasChangedGridSection) {
+      const originGrid = nextLayout[lastSectionId] as unknown as GridSectionData;
+      const resolvedOriginGrid = resolveGridSection(originGrid.panels);
+      (nextLayout[lastSectionId] as unknown as GridSectionData).panels = resolvedOriginGrid;
```

Comment:
these castings (`as`) are not needed :)

Reply:
Oops! I had these castings everywhere in an earlier version of the code **before** I changed some types around to make them work better, but I clearly missed some in my cleanup. Sorry about that - thanks for catching it 🙇  

fixed in c0a7c7cd3b0342c8c8a72f4542bd8bd496c59048